### DISCUSSION TITLE
What does "private" email address mean in context of GitHub web and Git commands?

### DISCUSSION THREAD

**Discussion by lusrx** [community/community#22762] [New to GitHub]
<p>Hello,</p>
<p>At <code>https://github.com/settings/emails</code>, I checked the two boxes “Keep my email addresses private” and “Block command line pushes that expose my email”. Then I got a notification saying:</p>
<blockquote>
<p>“Commits pushed with a private email will now be blocked and you will see a warning.”</p>
</blockquote>
<p>What is “private” in this context?</p>
<ul>
<li>A: <code>john.doe@mail.com</code>
</li>
<li>B: <code>31415926+tesla@users.noreply.github.com</code>
</li>
</ul>
<p>Is it A or is it B?</p>
<p><strong>Q:</strong> Does this mean that if <code>user.email</code> in Git config file is set to A I will not be able to push to the repo on GitHub?</p>
<p><strong>Q:</strong> In other words, if “Block command line pushes …” is set, I need to make sure that my Git config uses email B to ensure I can still push to the repo on GitHub?</p>
<p><strong>Q:</strong> Also, when I check “Keep my email addresses private”, does this mean that the GitHub noreply email address (B) will be used when using GitHub web interface?</p>
<p><strong>Q:</strong> When using GitHub web interface, where are these email addresses exposed? Despite having one public repo on GitHub I can’t find any place where this email is listed.</p>
<p>Sorry for these noob questions, I’m new to Git and to GitHub. Just trying to understand how to keep my email address private and still be able to use and learn Git from the command line and from GitHub web interface.</p>
<p>Regards,<br>
lusrx</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Great questions :grinning:</p>
<blockquote>
<p>What is “private” in this context?</p>
</blockquote>
<p><strong>A</strong> is the “private” email, in other words, the email that you don’t want exposed publicly.</p>
<blockquote>
<p>Does this mean that if  <code>user.email</code>  in Git config file is set to A I will not be able to push to the repo on GitHub?</p>
</blockquote>
<p>Yes, that’s correct.</p>
<blockquote>
<p>In other words, if “Block command line pushes …” is set, I need to make sure that my Git config uses email B to ensure I can still push to the repo on GitHub?</p>
</blockquote>
<p>Yes, that’s correct.</p>
<blockquote>
<p>Also, when I check “Keep my email addresses private”, does this mean that the GitHub noreply email address (B) will be used when using GitHub web interface?</p>
</blockquote>
<p>Yes, that’s correct.</p>
<blockquote>
<p>When using GitHub web interface, where are these email addresses exposed? Despite having one public repo on GitHub I can’t find any place where this email is listed.</p>
</blockquote>
<p>I’m not sure what you’re asking here. Can you be more specific as to what you mean exactly?</p>
<p>Let us know!</p>

**Comment by lusrx** [comment]
<p>Thanks!</p>
<p>You know, sometimes I doubt myself and seek confirmation from others, even though I seem to understand things fairly well (English is not my first language). It takes time, experience and conviction to accept the truth. But some things are just so complex to wrap your head around, they are not always easy to test. Which is why it’s great that we have fantastic people in the community who can help each other out! :hugs:</p>
<aside class="quote group-github-employees" data-username="lee-dohm" data-post="2" data-topic="135818">
<div class="title">
<div class="quote-controls"></div>
 lee-dohm:</div>
<blockquote>
<p>I’m not sure what you’re asking here. Can you be more specific as to what you mean exactly?</p>
</blockquote>
</aside>
<p>I am trying to understand the visibility level or impact level. In other words, if I decide against keeping my “private” email (A) private, who can see it and where? I think I have seen these emails listed in git command line when I look at the log, but I only see usernames when looking at the commit history on GitHub.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="lusrx" data-post="3" data-topic="135818">
<div class="title">
<div class="quote-controls"></div>
 lusrx:</div>
<blockquote>
<p>I think I have seen these emails listed in git command line when I look at the log, but I only see usernames when looking at the commit history on GitHub.</p>
</blockquote>
</aside>
<p>Github showing usernames in the web UI is only a UI thing, it doesn’t change anything about the repository data. If anyone clones the repository and looks at the log or looks at the raw commits over the web interface (try attaching <code>.patch</code> to any commit URL), they will see the name and mail address in the commit, not your Github username.</p>
